applications/zigbee_weather_station/src/weather_station.h

```diff
@@ -9,6 +9,44 @@
 
 #include <zcl/zb_zcl_temp_measurement_addons.h>
 
+#include "sensor.h"
+
+/* Zigbee Cluster Library 4.4.2.2.1.1: MeasuredValue = 100x temperature in degrees Celsius */
+#define ZCL_TEMPERATURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER 100
+/* Zigbee Cluster Library 4.5.2.2.1.1: MeasuredValue = 10x pressure in kPa */
+#define ZCL_PRESSURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER 10
+/* Zigbee Cluster Library 4.7.2.1.1: MeasuredValue = 100x water content in % */
+#define ZCL_HUMIDITY_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER 100
+
+/* Measurements ranges scaled for attribute values */
+#define WEATHER_STATION_ATTR_TEMP_MIN (	  \
+		SENSOR_TEMP_CELSIUS_MIN * \
+		ZCL_TEMPERATURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_TEMP_MAX (	  \
+		SENSOR_TEMP_CELSIUS_MAX * \
+		ZCL_TEMPERATURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_TEMP_TOLERANCE (	\
+		SENSOR_TEMP_CELSIUS_TOLERANCE *	\
+		ZCL_TEMPERATURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_PRESSURE_MIN ( \
+		SENSOR_PRESSURE_KPA_MIN *   \
+		ZCL_PRESSURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_PRESSURE_MAX ( \
+		SENSOR_PRESSURE_KPA_MAX *   \
+		ZCL_PRESSURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_PRESSURE_TOLERANCE ( \
+		SENSOR_PRESSURE_KPA_TOLERANCE *	  \
+		ZCL_PRESSURE_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_HUMIDITY_MIN (   \
+		SENSOR_HUMIDITY_PERCENT_MIN * \
+		ZCL_HUMIDITY_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_HUMIDITY_MAX (   \
+		SENSOR_HUMIDITY_PERCENT_MAX * \
+		ZCL_HUMIDITY_MEMASUREMENT_MEASURED_VALUE_MULTIPLIER)
+#define WEATHER_STATION_ATTR_HUMIDITY_TOLERANCE (   \
```

Comment:
In src/main.c, line 140 it's mentioned that humidity's tolerance is not supported at the moment. I would suggest to remove this define if redundant :)

Reply:
Yes, but humidity tolerance is not supported on ZCL cluster level whereas sensor supports it.
I left it intentionally, especially that there is a task to add support for humidity tolerance: https://projecttools.nordicsemi.no/jira/browse/KRKNWK-13601